### ISSUE TITLE
Cannot get tagged/all documents from Bucket

### ISSUE THREAD

**Issue by Yamakasi** [PHPSocialNetwork/phpfastcache#633]
I'm trying to get tagged documents from my Bucket but I get back and empty array. When I try to get all documents like:

` $CachedString = $this->InstanceCache->getItems();`

I also get an empty array back.

->getConfig(); gives me the right config of the bucket I query, I also tried to get these tagged documents on key but that didn't got me any result as well. Is there something different when you tag the documents or can I debug this in a better way ?

**Comment by Yamakasi** [comment]
Yes I read that and could not make a clue out of it, the same for getting them by tags.

**Comment by Geolim4** [comment]
Tags are exclusive to Phpfastcache it's handled by `ExtendedCacheItemPoolInterface`.
So ATM the lib is working welln I highly encourage your to use a true IDE and not a simple text editor.

An IDE (such as Phpstorm) will highlight your code type hints,  or highlight you what's a method is supposed to return, etc.

**Comment by Yamakasi** [comment]
I'm using Netbeans, it's a good IDE which I use for years. It was a question, not a bug... you might saw the wrong tag "Simple question" ;)

**Comment by Geolim4** [comment]
The question is actually solved, it's just that I don't want to get a whole stack of dead tickets like many projects has today. It's simply about organization since I'm almost alone to handle the issues actually.

**Comment by Yamakasi** [comment]
I understand, I mostly report back, even by email if needed, we are not all the same. Issue seems to be solved as I think the documents got expired and my admin panel didn't refresh.

During my searching (which I do a lot) I could not find out if we can get items back in groups of arrays so like 

`[['item1tag1, item1tag2, item1tag3],['item2tag1, item2tag2, item2tag3]]`

And we get only back item1 and item2 or do I need to get per item ?

**Comment by Geolim4** [comment]
Tags documentation is listed here:
https://github.com/PHPSocialNetwork/phpfastcache#itempool-api-extendedcacheitempoolinterface

You maybe interested by
`getItemsByTags(array $tagNames)` _Returns items by one of multiple tag names_
or
`getItemsByTagsAll($tagName)` _Returns items by all of multiple tag names_
